_core/smart/tutorial-counter.md

```diff
@@ -0,0 +1,193 @@
+
+1. Using your terminal, run the following command:
+
+    ```bash
+    npm init clarity-starter
+    ```
+
+    You have to select a template and a name for your local folder. For the counter template used in this tutorial, ensure to type `counter` and hit ENTER:
+
+    ```bash
+    ? Template - one of [hello-world, counter]: counter
+    ? Project name: (clarity-counter-project)
+    ```
+
+    Finally, the project dependencies are installed and your project is ready for development. Because you already completed the Hello World tutorial, the project structure is familiar to you. The main difference is that we have additional tests for a new counter smart contract.
+
+## Step 2: Running tests
+
+Smart contracts are often developed in a test-driven approach to ensure code quality but also to speed up the development cycle by removing the need to push every change to the Blockchain before executing it. We will do the same in this project. Now, let's run the tests and review the results:
+
+1. Still in the project root directory, run the following command:
+
+    ```bash
+    npm test
+    ```
+
+    You should see the following response:
+
+    ```bash
+    counter contract test suite
+        ✓ should have a valid syntax
+        deploying an instance of the contract
+        1) should start at zero
+        2) should increment
+        3) should decrement
+
+    hello world contract test suite
+        ✓ should have a valid syntax
+        deploying an instance of the contract
+        ✓ should return 'hello world'
+        ✓ should echo number
+
+
+    4 passing (455ms)
+    3 failing
+    ```
+
+    It looks like we see some failed test! That is on purpose - we will implement the new smart contract in the next steps! After every increment of the contract, we will rerun the tests to ensure we're on the right track.
+
+## Step 3: Developing a smart contract
+
+Let's get familiar with the tests to understand what the new smart contract should look like
+
+1. Take a quick look at the test file associated with the counter smart contract:
+
+    ```shell
+    cat test/counter.ts
+    ```
+
+    You should be familiar with the test set up from the Hello World tutorial. Notice how the instance of the smart contract is created on line 8:
+
+    ```js
+    counterClient = new Client("SP3GWX3NE58KXHESRYE4DYQ1S31PQJTCRXB3PE9SB.counter", "counter", provider);
+    ```
+
+    That tells us that the new smart contract is named `counter` and that we need to start by creating a new file for the smart contract: `contracts/counter.clar`. Note that the `contracts` folder is assumed as the base folder and that every Clarity file has the suffix `.clar`.
+
+2. Let's create the new file:
+
+    ```shell
+    touch contracts/counter.clar
+    ```
+
+3. With the editor of your choice, open the file and add the following lines of code:
+
+    ```cl
+    (define-data-var counter int 0)
```

Comment:
The indentation here is throwing things off, and copy/paste results in wacky indentation. There is no need to indent the whole code block, the 3 backticks is all that is required.

Reply:
Tried that, but it creates a but with the numbering (3. will become 1. at every paragraph)

Reply:
D'oh... yeah that makes sense. Can we try wrapping the line in `<p>..</p>`?